playwright-examples/src/test/java/com/saucelabs/bindings/SaucePlaywrightSession.java

```diff
@@ -0,0 +1,133 @@
+package com.saucelabs.bindings;
+
+import com.microsoft.playwright.APIRequest;
+import com.microsoft.playwright.APIRequestContext;
+import com.microsoft.playwright.APIResponse;
+import com.microsoft.playwright.Browser;
+import com.microsoft.playwright.Playwright;
+import com.microsoft.playwright.options.RequestOptions;
+import com.saucelabs.saucebindings.DataCenter;
+import com.saucelabs.saucebindings.SauceRest;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Collections;
+import java.util.Map;
+import lombok.Getter;
+import lombok.Setter;
+import org.json.JSONObject;
+import org.openqa.selenium.InvalidArgumentException;
+import org.openqa.selenium.remote.SessionId;
+
+public class SaucePlaywrightSession {
+  @Getter protected Playwright playwright;
+  @Getter @Setter private DataCenter dataCenter = DataCenter.US_WEST;
+  private final Map<String, Object> capabilities;
+  @Setter private URL sauceUrl;
+  @Getter private Boolean result;
+  private String id;
+  private String cdpEndpoint;
+  private SauceRest rest;
+  private APIRequestContext request;
+  private Browser browser;
+
+  public SaucePlaywrightSession(Map<String, Object> capabilities) {
+    this.capabilities = capabilities;
+  }
+
+  public Browser start() {
+    this.browser = createPlaywrightSession();
+
+    this.rest = sauceRest();
+    return browser;
+  }
+
+  SauceRest sauceRest() {
+    return new SauceRest(this.dataCenter, new SessionId(id));
+  }
+
+  private Browser createPlaywrightSession() {
+    this.playwright = Playwright.create();
+    this.request =
+        playwright
+            .request()
+            .newContext(
+                new APIRequest.NewContextOptions().setBaseURL(String.valueOf(getSauceUrl())));
+
+    JSONObject jsonPayload = new JSONObject(capabilities);
+    RequestOptions options =
+        RequestOptions.create()
+            .setMethod("POST")
+            .setData(jsonPayload.toString())
+            .setMaxRedirects(5)
+            .setTimeout(120000);
+    APIResponse newSessionResponse = request.fetch("session", options);
+    JSONObject response = new JSONObject(newSessionResponse.text()).getJSONObject("value");
+
+    this.id = response.getString("sessionId");
+    this.cdpEndpoint = response.getJSONObject("capabilities").getString("se:cdp");
```

Comment:
Does this fail gracefully, or should there be some exception handling?

Reply:
This is mostly proof of concept code that I wanted to put *somewhere* would need more love if we put it in a library.